File: C/Algorithms/2220.c
int minBitFlips(int start, int goal){
    int count = 0, b, s;
    if (start > goal) {
        b = start, s = goal;
    } else {
        b = goal, s = start;
    }
    
    while (s) {
        if ((s & 1) != (b & 1))
            count++;
        s >>= 1;
        b >>= 1;
    }
    while (b) {
        if (b & 1)
            count++;
        b >>= 1;
    }
    return count;
}

File: Python/Algorithms/2220.py
class Solution:
    def minBitFlips(self, start: int, goal: int) -> int:
        count = 0
        (b, s) = (start, goal) if start > goal else (goal, start)
        while s:
            if b & 1 != s & 1:
                count += 1
            b >>= 1
            s >>= 1
        while b:
            if b & 1:
                count += 1
            b >>= 1
        return count